### PR TITLE
Reduce RealNumberQ in builtin.image.basic and ...

### PR DESCRIPTION
**Pull request by rocky**
Disallow Python 3.6
move is_integer_rational_or_real to mathics.core.atoms

### CODE REVIEW

File: mathics/builtin/image/basic.py [hunk 1]
```diff
@@ -40,12 +46,15 @@ class Blur(Builtin):
      = -Image-
     """
 
-    summary_text = "blur an image"
     rules = {
         "Blur[image_Image]": "Blur[image, 2]",
-        "Blur[image_Image, r_?RealNumberQ]": "ImageConvolve[image, BoxMatrix[r] / Total[Flatten[BoxMatrix[r]]]]",
+        "Blur[image_Image, r_?RealNumberQ]": (
```
Comment (rocky) [inline mathics/builtin/image/basic.py:51]: I haven't been able to figure out the replacement pattern here.

Comment (mmatera) [reply mathics/builtin/image/basic.py:51]: One possibility would be    `r:(_Integer|_Real|_Rational)`. On the other hand, we can keep  `RealNumberQ` but put it in a private context.

Comment (mmatera) [reply mathics/builtin/image/basic.py:51]: Here is another one: `r_/;(NumberQ[r] && (Im[r]==0))`. Also we can just ask for NumberQ, and then in the "body" check `r` has a nonzero imaginary part and send a message. Then pass `BoxMatrix` the real part.

Comment (mmatera) [reply mathics/builtin/image/basic.py:51]: Still, there is another one: `r_?NumberQ`  and then define another rule specific for Complex numbers, that just show the warning.

In any case, I wonder why there is no keyword in WL that allows having the behavior we obtain when we implement the rule as a method,  and the method returns (Python's) `Null`.

Comment (rocky) [reply mathics/builtin/image/basic.py]: One thing I am coming to understand is that this rewrite-rule approach generally leads to different behavior than WMA. 

First, when there is no match, Mathics3 silently fails whereas in WMA a type error message is given.

Second, when there is an error detected a rewrite may have already occurred leading to a result that was different from the original input given. For example If I write; 

```
ImageAdjust[x, "a"]
```

By the time the message occurs that "a" is invalid, This has already been rewritten to `ImageAdjust[x, {"a", 0, 1}]` and this is confusing. 

While on the one hand it is nice that there was an attempt to use more Mathics3 as opposed to Python, on the other hand it leads to less WMA conformance, and more user confusion. 

So again, to me this is another example of "too clever by a half".

**Comment by rocky** [comment]
Merging and iterating...

**Pull request merged**